notification/slack/channel.go

```diff
@@ -248,6 +248,9 @@ func (s *ChannelSender) loadChannels(ctx context.Context) ([]Channel, error) {
 		}
 
 		if !resData.OK {
+			acceptedScopes := resp.Header.Get("X-Accepted-Oauth-Scopes")
+			providedScopes := resp.Header.Get("X-Oauth-Scopes")
+			log.Log(ctx, errors.New("Slack app scopes must include one of: ["+acceptedScopes+"]; got: ["+providedScopes+"]"))
```

Comment:
Let's make this a little more specific, and only log the scope info IF it's a scope error
```suggestion
		if !resData.OK {
			if resData.Error == "missing_scope" {
				acceptedScopes := resp.Header.Get("X-Accepted-Oauth-Scopes")
				providedScopes := resp.Header.Get("X-Oauth-Scopes")
				log.Log(ctx, fmt.Errorf("list Slack channels: missing_scope; need one of %v but got %v", acceptedScopes, providedScopes))
			}
```

Reply:
Though, it might be better to pass `resp.Header` to `wrapError` and put this code inside the `missing_scope` case instead of here -- that way we log missing scope error for ALL potential slack calls, and don't have two places we're handling `missing_scope`.

Reply:
Yeah, a more proper way would probably be to make a custom error type we can embed the data with, then handle logging/mapping of the error as part of sanitizing (e.g. see how `sqlutil.MapError` or `errutil.HTTPError` work). The one difference here is we want to have a specific message for the user and in this one special case log the config issue.

```go
type apiError struct {
	msg string
	headers http.Header
}
func (err apiError) Error() string{return err.msg}
func mapError(ctx context.Context, err error)error {
	var apiErr apiError
	if !errors.As(err, &apiErr) {
		log.Log(ctx,err)
		return slackError("unexpected error")
	}

	switch apiErr.msg {
	case "missing_scope":
		log.Log(ctx, fmt.Errorf(/*... apiErr.headers ...*/))
		return slackError("Permission Denied")
	}

	log.Log(ctx, err)
	return slackError("unexpected error")
}

func loadChannel() {
	return nil,  fmt.Errorf("list slack channels: %w",&apiError{msg: respData.Error, headers: resp.Headers})
}
func Channel() {
	foo,err:=loadChannel()
	if err != nil {
		return nil, mapError(ctx, err)
	}
}
```

That being said, I wouldn't worry much about it just being a switch statement, it's more about keeping ownership and concerns separate. I think a custom error type would be better than adding a bunch of arguments though.